devicetypes/sensative/sensative-strips-guard-700.src/sensative-strips-guard-700.groovy

```diff
@@ -0,0 +1,482 @@
+
+		attribute "lastCheckIn", "string"
+
+		fingerprint mfr:"019A", prod:"0004", model:"0004", deviceJoinName: "Strips Guard 700"
+	}
+
+	simulator { }
+
+	preferences {
+		configParams.each { param ->
+			input "configParam${param.num}", "enum",
+				title: "${param.name}:",
+				required: false,
+				displayDuringSetup: false,
+				options: param.options
+		}
+
+		input "debugLogging", "enum",
+			title: "Logging:",
+			required: false,
+			defaultValue: 1,
+			options: [0:"Disabled", 1:"Enabled [DEFAULT]"]
+	}
+}
+
+
+def installed() {
+	logDebug "installed()..."
+	state.pendingRefresh = true
+	initialize()
+}
+
+def updated() {
+	if (!isDuplicateCommand(state.lastUpdated, 1000)) {
+		state.lastUpdated = new Date().time
+
+		logDebug "updated()..."
+		
+		initialize()
+
+		if (pendingChanges) {
+			logForceWakeupMessage("The configuration changes will be sent to the device the next time it wakes up.")
+		}
+	}
+}
+
+void initialize() {
+	state.debugLoggingEnabled = (safeToInt(settings?.debugOutput, 1) != 0)
+	
+	if (!device.currentValue("tamper")) {
+		sendEventIfNew("tamper", "clear")
+	}
+	
+	if (!device.currentValue("checkInterval")) {
+		sendEvent(name: "checkInterval", value: ((wakeUpIntervalSeconds * 2) + 300), displayed: falsle, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
+	}
+}
+
+
+def configure() {
+	logDebug "configure()..."
+
+	state.pendingRefresh = true
+	sendCommands(getConfigureCmds())
+}
+
+List<String> getConfigureCmds() {
+	runIn(6, refreshSyncStatus)
+
+	int changes = pendingChanges
+	if (changes) {
+		log.warn "Syncing ${changes} Change(s)"
+	}
+
+	List<String> cmds = [ ]
+
+	if (state.pendingRefresh) {		
+		cmds << batteryGetCmd()
+		cmds << versionGetCmd()
+		cmds << sensorBinaryGetCmd()
+	}
+	
+	if (state.pendingRefresh || (state.wakeUpInterval != wakeUpIntervalSeconds)) {
+		logDebug "Changing wake up interval to ${wakeUpIntervalSeconds} seconds"
+		cmds << wakeUpIntervalSetCmd(wakeUpIntervalSeconds)
+		cmds << wakeUpIntervalGetCmd()
+	}
+
+	configParams.each {
+		Integer storedVal = getParamStoredValue(it.num)
+		Integer settingVal = getSettingValue(it.num)
+		
+		if ((settingVal != null) && (settingVal != storedVal)) {
+			logDebug "CHANGING ${it.name}(#${it.num}) from ${storedVal} to ${settingVal}"
+			cmds << configSetCmd(it, settingVal)
+			cmds << configGetCmd(it)
+		}
+		else if (state.pendingRefresh) {
+			cmds << configGetCmd(it)
+		}
+	}
+
+	state.pendingRefresh = false
+	return cmds
+}
+
+
+// Required for HealthCheck Capability, but doesn't actually do anything because this device sleeps.
+def ping() {
+	logDebug "ping()"
+}
+
+
+def refresh() {
+	logDebug "refresh()..."
+
+	state.pendingRefresh = true
+	logForceWakeupMessage("The device will be refreshed the next time it wakes up.")	
+}
+
+void logForceWakeupMessage(String msg) {
+	log.warn "${msg}  To force the device to wake up immediately, move the magnet towards the round end 3 times."
+}
+
+
+String wakeUpIntervalGetCmd() {
```

Comment:
Since this method is called only in one place, I suggest to inline this code instead of having this method.

Reply:
Made change